models/issue_dependency.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"time"
+)
+
+// IssueDependency is connection request for receiving issue notification.
+type IssueDependency struct {
+	ID           int64     `xorm:"pk autoincr"`
+	UserID       int64     `xorm:"UNIQUE(watch) NOT NULL"`
+	IssueID      int64     `xorm:"UNIQUE(watch) NOT NULL"`
+	DependencyID int64    `xorm:"UNIQUE(watch) NOT NULL"`
+	Created      time.Time `xorm:"-"`
+	CreatedUnix  int64     `xorm:"NOT NULL"`
+	Updated      time.Time `xorm:"-"`
+	UpdatedUnix  int64     `xorm:"NOT NULL"`
+}
+
+// BeforeInsert is invoked from XORM before inserting an object of this type.
+func (iw *IssueDependency) BeforeInsert() {
+	var (
+		t = time.Now()
+		u = t.Unix()
+	)
+	iw.Created = t
+	iw.CreatedUnix = u
+	iw.Updated = t
+	iw.UpdatedUnix = u
+}
+
+// BeforeUpdate is invoked from XORM before updating an object of this type.
+func (iw *IssueDependency) BeforeUpdate() {
+	var (
+		t = time.Now()
+		u = t.Unix()
+	)
+	iw.Updated = t
+	iw.UpdatedUnix = u
+}
+
+// CreateIssueDependency creates a new dependency for an issue
+func CreateIssueDependency(user *User, issue, dep *Issue) (err error, exists bool, depExists bool) {
+	sess := x.NewSession()
+
+	// TODO: Move this to the appropriate place
+	err = x.Sync(new(IssueDependency))
```

Comment:
Please move this to a migration.

Reply:
As i stated above:
> There is some work to be done with migrations etc, currently i simply put the xorm-call at the beginning of the function, which probably shouldn't be there. Where is the appropriate place to put this?

Where?

Reply:
Migrations are found in models/migrations. Create a new one and call from migrations.go.